### PR TITLE
chore(admin): add audit-logs contracts

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* adds API contracts for audit-logs

### Why is it needed?

* so we can type audit-logs

### CODE REVIEW

File: packages/core/admin/shared/contracts/audit-logs.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { Entity, SanitizedAdminUser } from './shared';
+import { Schema } from '@strapi/types';
+
+interface CollectionAuditLog extends Schema.CollectionType {
+  user?: SanitizedAdminUser;
+}
+
+interface SingleAuditLog extends Schema.SingleType {
+  user?: SanitizedAdminUser;
+}
```
Comment (Marc-Roig) [inline packages/core/admin/shared/contracts/audit-logs.ts:11]: doesnt the audit log also return the payload of the log?

File: packages/core/admin/shared/contracts/audit-logs.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+namespace GetAll {
+  export interface Request {
+    body: {};
+    query: {};
+  }
+
+  export interface Response {
+    data: AuditLog[];
+    error?: errors.ApplicationError;
+  }
```
Comment (remidej) [inline packages/core/admin/shared/contracts/audit-logs.ts:24]: This doesn't seem to match what I see when calling the endpoint, which is more like

```ts
interface Response {
  pagination: {
    page: number;
    pageSize: number;
    pageCount: number;
    total: number;
  };
  results: AuditLog[];
}
```

error is probably right though I haven't tested

File: packages/core/admin/shared/contracts/audit-logs.ts (hunk 1 above)
Comment (joshuaellis) [reply packages/core/admin/shared/contracts/audit-logs.ts:11]: Isn't that what the schema is? 😬 or is this not an entity?

Comment (remidej) [inline packages/core/admin/shared/contracts/audit-logs.ts:11]: I'm not sure this is right, from what I see in my API response the audit log shape is very simple: only id, action, date, payload and user keys. Not attributes object or anything else

Also I think User is mandato

Comment (Marc-Roig) [reply packages/core/admin/shared/contracts/audit-logs.ts:11]: oops, I didnd see the extends :D

Comment (joshuaellis) [reply packages/core/admin/shared/contracts/audit-logs.ts:11]: So what would that look like?

```ts
interface AuditLog {
	id: Entity.ID;
	date: string;
	action: string; // is this a union of specific actions?
	payload: unknown;
	user: SanitizedAdminUser
}
```

What is a `user key`? and what is the `payload? 🤔

Comment (remidej) [reply packages/core/admin/shared/contracts/audit-logs.ts:11]: Yes that looks good.

- user is the sanitized object 👍 it's now effectively required, but it wasn't when we launched the feature, so I think we have to leave it as optional
- action: currently we know the list so a union would work. But [we planned](https://github.com/strapi/strapi/blob/main/packages/core/admin/ee/server/services/audit-logs.js#L88-L89) on later letting users save custom events too. So I would leave it as a string
- payload depends on each action type, so I think unknown makes sense.

Comment (remidej) [reply packages/core/admin/shared/contracts/audit-logs.ts:11]: we do know that payload is json actually

Comment (joshuaellis) [reply packages/core/admin/shared/contracts/audit-logs.ts:11]: Okay, let me make some changes and you can review again :)

File: packages/core/admin/shared/contracts/audit-logs.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  export interface Params {
+    id: Entity['id'];
+  }
+
+  export interface Response {
+    data: AuditLog;
+    error?: errors.ApplicationError;
+  }
```
Comment (remidej) [inline packages/core/admin/shared/contracts/audit-logs.ts:43]: I hadn't checked the single response before sorry. There's no data key, it's directly the AuditLog object

Comment (joshuaellis) [reply packages/core/admin/shared/contracts/audit-logs.ts:43]: Really? What happens if it errors?

Comment (joshuaellis) [reply packages/core/admin/shared/contracts/audit-logs.ts:43]: @Marc-Roig any idea how to type this case? Should `Reponse` be a union `AuditLog | errors.ApplicationError`?

Comment (Marc-Roig) [reply packages/core/admin/shared/contracts/audit-logs.ts:43]: yep, I think this also interests @jhoward1994 , you find a similar case and we were asking about the same thing. We could do

export type Response = AudigLog | { error: errors.ApplicationError }

**Review by remidej — APPROVED** [review]

**Comment by joshuaellis** [comment]
Thanks for all the help!

**Pull request merged**